internal/app/decision_maker.go

```diff
@@ -60,18 +60,115 @@ func (cs *currentState) makePlan(s *state) *plan {
 
 	wg := sync.WaitGroup{}
 	sem := make(chan struct{}, resourcePool)
+	namesC := make(chan [2]string, len(s.Apps))
+	versionsC := make(chan [4]string, len(s.Apps))
+
+	// We store the results of the helm commands
+	extractedChartNames := make(map[string]string)
```

Comment:
probably using a struct for the extracted chart names and versions would be cleaner but I guess we can do that in another iteation